web/src/compose.js

```diff
@@ -507,6 +509,21 @@ export function initialize() {
         },
     );
 
+    $("body").on(
+        "click",
+        `.${CSS.escape(
+            compose_banner.CLASSNAMES.compose_in_search_view,
+        )} .main-view-banner-action-button`,
+        (event) => {
+            event.preventDefault();
+
+            const filter = new Filter(narrow_state.operators());
+            window.location.href = filter.generate_redirect_url();
```

Comment:
You haven't updated this part yet -- see my previous comment about this not working for plain text queries.